src/manager/node.c

```diff
@@ -174,6 +191,14 @@ void node_unref(Node *node) {
                 proxy_dependency_free(dep);
         }
 
+        Subscription *sub = NULL;
+        Subscription *next_sub = NULL;
+        LIST_FOREACH_SAFE(wildcard_subscriptions, sub, next_sub, node->wildcard_subscriptions) {
+                manager_remove_subscription(node->manager, sub);
+                LIST_REMOVE(wildcard_subscriptions, node->wildcard_subscriptions, sub);
+                subscription_unref(sub);
```

Comment:
This part doesn't look right to me. Yes, we should maybe remove the sub from wildcard_subscriptions, because node_subscribe_wildcard_unit() added it there. However, we should not unref it, because we didn't ref it.  Alterntively we should ref the sub when adding it, but I don't think that is right, as we risk creating cycles. And anyway the node will not be unref:ed until shutdown, so lifetimes are not really a problem.

And we shouldn't call manager_remove_subscription(), which will remove the sub from other nodes too. That should only happen when the subscription is unsubscribed, which comes from the monitor.

Reply:
Removed this part completely since I merged the `wildcard_subscriptions` into the existing hashmap for the `unit_subscriptions`.